### PR TITLE
[ORPO] Update NLL loss to use `input_ids` instead

### PR DESCRIPTION
**Pull request by alvarobartt**
## Description

This PR updates how the NLL loss in ORPO is being calculated so as to use the `input_ids` for the chosen response i.e. the whole conversation, instead of only the labels i.e. the ids for the chosen response only.

As mentioned by @nlee-208, calculating the full prompt loss instead of solely the loss on the chosen response leads to better results in MT-Bench, leading then to a more faithful reproduction of the original codebase at https://github.com/xfactlab/orpo.git (see the screenshot shared by Noah below where the `wo-pl`, standing for without prompt loss, fine-tune performs worst on MT-Bench)

image

Kudos to @nlee-208, @jiwooya1000 and @lewtun

### CODE REVIEW

**Review by kashif — APPROVED** [review]

**Comment by kashif** [comment]
@alvarobartt so the failing tests are with regards to the enc-decoder models I believe

**Comment by alvarobartt** [comment]
> @alvarobartt so the failing tests are with regards to the enc-decoder models I believe

Sure, let me check those!

**Review by lewtun — APPROVED** [review]
Thanks for the fix! LGTM but let's wait until we've figure out why the CI is red on `main` before merging this

**Review by kashif — APPROVED** [review]

**Comment by lewtun** [comment]
Chatted with @younesbelkada about the failing slow tests (seems like a timeout issue), so it's fine to merge this

**Pull request merged**
